### ISSUE TITLE
Donkey kong 64 doesn't boot up

### ISSUE THREAD

**Issue by richard42** [mupen64plus/mupen64plus-user-issues#272]
Originally reported on Google Code with ID 273

```
Describe your system:
 - Linux distribution: Debian
 - Machine type: 64-bit
 - Mupen64Plus version: svn1404 + bug 51 related patches
 - Plugins used: glide64

For visual artifacts, give:
 - MD5 118E9CE360B97A6F8DAB2C9F30660BF5
 - Donkey Kong 64 (E) [!]

Describe the problem:
It seems that the command at address 0x80603fb0 or following command (in
delay slot) isn't interpreted correctly and it causes Donkey Kong 64 not to
boot up correctly. This is a 'bne  $at, $r0, 0x80603FA4'

It seems to be inside a loop "forever" near that position. The code looks
like this:

80603FA4 lw    $t7, 0x0000($v0)
80603FA8 addiu $v0, 0x0004
80603FAC sltu  $at, $v0, $a0
80603FB0 bne   $at, $r0, 0x80603FA4
80603FB4 addu  $v1, $v1, $t7
80603FB8

And after the next round it will do an exception with cause 0x10000400.

I cannot upload it here as "project attachment quota exceeded".
http://www.mediafire.com/?dldyazngymn



```

Reported by `sven@narfation.org` on 2009-09-26 23:29:58

**Comment by richard42** [comment]
```
Sry, this savegame is exact one round before the actual interessting bne. It seems
that the interrupt by itself is the problem. Somehow.
```

Reported by `sven@narfation.org` on 2009-09-27 22:57:50

**Comment by richard42** [comment]
```
Pokemon Puzzle League 000364BAC80E41D9060A31A5923874B7 doesn't boot up either and
also creates many, many, many SP_INT interrupts instead. Maybe this is related somehow.
```

Reported by `sven@narfation.org` on 2009-09-28 10:52:11

**Comment by richard42** [comment]
```
bump

I really want to play DK on my mac yall
```

Reported by `firstdogmusic` on 2009-10-05 08:04:29

**Comment by richard42** [comment]
```
srsly

i <3 this game

but i havent played in 6 years

help a dog out
```

Reported by `firstdogmusic` on 2009-10-05 08:05:30

**Comment by richard42** [comment]
```
what the petes sake bro

its been like 18 hours

somebody help me
```

Reported by `firstdogmusic` on 2009-10-05 23:29:06

**Comment by richard42** [comment]
```
Bugs doesn't get fixed by writing stupid entries to a bug tracker.
```

Reported by `sven@narfation.org` on 2009-10-06 08:10:49

**Comment by richard42** [comment]
```
Corrected savestate http://www.mediafire.com/?0jbfwixgmyi
```

Reported by `sven@narfation.org` on 2009-10-08 18:48:31

**Comment by richard42** [comment]
Reported by `richard42g` on 2010-01-12 22:48:32
- **Labels added**: Type-Defect, Component-Core

**Comment by richard42** [comment]
```
Don't take my comment too seriously, because I just don't know, but may this be
caused by some division-by-zero strange behaviour? In the Mupen64Plus version 1.5 I
had, I got some division-by-zero messages just after the Rare logo (IIRC).
Unfortunately, after I compiled from the source, I didn't get those errors anymore
for some strange reason.

But I tried adding some checks to DIV_S for nearly-zero values, and at almost the end
of the intro I get one place where it's very close to zero, twice.

DIV_S 0.000003 @ 806545CC

I don't know if Mupen64Plus is accurate in floating point calculations, so if it
isn't, that might be an interesting place to look.
```

Reported by `GameZelda2` on 2010-05-17 21:06:12

**Comment by richard42** [comment]
```
This rom just needs to use 4k eeprom. I will try to merge the wii64 stuff with
mupen64plus.ini later
```

Reported by `sven@narfation.org` on 2010-06-03 21:14:56

**Comment by richard42** [comment]
```
This specific problem should be fixed in 196:24b506da4250
```

Reported by `sven@narfation.org` on 2010-06-03 22:41:25

**Comment by richard42** [comment]
Reported by `richard42g` on 2010-06-04 02:04:51
- **Status changed**: `Fixed`
